docs/Build.md

```diff
@@ -28,8 +28,10 @@ See the project's `README` for detailed information about installing `hie`.
 The build script `install.hs` defines several targets using the `shake` build system. The targets are roughly:
 
 * `hie-*`: builds and installs the `hie` binaries. Also renames the binaries to contain the correct version-number.
-* `build`: builds and installs `hie` binaries for all supported `ghc` versions.
+* `build-lastest`: builds ad installs `hie` for the last available and supported `ghc` version.
+* `build-all`: builds and installs `hie` binaries for all supported `ghc` versions. This option may take a long time and computer resources so use it with caution.
```

Comment:
s/computer resources so use it with caution/a lot of cpu resources, so use with caution/ maybe?

Reply:
it consumes a lot of memory too :thinking:

Reply:
Do we really need this target? It's confusing for newcomers and they're complaining about "hie takes ages to build and uses dozens of gigabytes of disk space". It's still a thing I'm seeing in reddit discussions.

Reply:
Sometimes you want it. But it is discouraged by the script from now on, meaning it is no longer the default "build" and displays a warning, iirc. Is it still confusing with these changes?

Reply:
Not sure. What are the use-cases? I imagine it's mostly for CI and binary packaging, right? If so, maybe add additional note like "Used for CI / development / packaging. You probably want a hie-latest or hie-X.Y.Z for your specific GHC"?

Reply:
Fair. I would actually assume that there currently is no real use-case. Except when someone *really* wants to install all supported ghc versions. We can probably remove it?

Reply:
> Except when someone really wants to install all supported ghc versions

Welp, that someone can make a trivial script like `./install.hs hie-8.4.4 hie-8.6.5 ...` 🤔 , so not much of a loss in case of removing this target. So looks like there is no concrete pros for this target, but we know at least one con.

Reply:
Maybe the main case would be ours: to test dev changes compiles and are functionally correct for all supported ghcs.
I dont know if someone is already using it (in ci or whatever) and their motivation, maybe we could add a deprecation note and remove it in the next version if nobody complains meanwhile?.

Reply:
Or change the target name to `build-all-available-ghcs` (as i did originally) and remove it in the next release

Reply:
We do not use it in CI, the arch AUR uses a custom build script, I really think nobody is using it.
Locally installing all versions also seems infeasible. What I normally do is let the CI verify that everything works for every version and if there is a bug for a specific GHC version, I install only that.

Reply:
ok, we can take the other way: remove it now to avoid unwanted full builds and add later if someone bring us a legitimate case use

Reply:
Yeah, I agree

Reply:
removed